### PR TITLE
Show new header on SIWE screen

### LINKED ISSUE #17812: New header is not shown on SIWE screen
## Background 
A new signature screen has been introduced for SIWE in https://github.com/MetaMask/metamask-extension/issues/17461, but it's missing the new header that we included in all signature requests here https://github.com/MetaMask/metamask-extension/issues/15705 . We should replace the PermissionsConnectHeader usage on this custom signature screen with the new header

## Steps to Reproduce
1. Go to https://metamask.github.io/test-dapp/
2. Under Sign In With Ethereum, click on Sign In With Ethereum

## Screenshot
Screenshot 2023-02-17 at 17 56 08

### PR DESCRIPTION
**Pull request by filipsekulic**
## Explanation
A new header is now being shown on the `SIWE` screen.

* Fixes #17812

### Before

https://user-images.githubusercontent.com/92310504/223081201-c2e3a21b-873e-4d73-9268-0a6a129b2614.mov

### After

https://user-images.githubusercontent.com/92310504/223075665-dce85014-f143-4979-9ef9-e666a4c14a61.mov

## Manual Testing Steps

1. Go to https://metamask.github.io/test-dapp/
2. Under Sign In With Ethereum, click on Sign In With Ethereum

### CODE REVIEW

**Comment by dzfjz** [comment]
Verified by QA

File: ui/components/app/signature-request-siwe/signature-request-siwe.js [hunk 1]
```diff
@@ -82,8 +92,45 @@ export default function SignatureRequestSIWE({
...
+    fromAccount.balance,
+    16,
+    EtherDenomination.WEI,
+  )
+    .toDenomination(EtherDenomination.ETH)
+    .toBase(10)
+    .round(6)
+    .toString();
+
+  const getNetworkName = () => {
```
Comment (darkwing) [inline ui/components/app/signature-request-siwe/signature-request-siwe.js:105]: I see this function exists elsewhere in the codebase a few times.  Would it be possible to move it to a helper file so we aren't repeating as much?  You'd pass the helper function `provider` and `t`

**Review by darkwing — CHANGES_REQUESTED** [review]

File: ui/components/app/dropdowns/network-dropdown.js [hunk 2]
```diff
@@ -227,27 +227,6 @@ class NetworkDropdown extends Component {
...
-        return t('sepolia');
-      case NETWORK_TYPES.LINEA_TESTNET:
-        return t('lineatestnet');
-      case NETWORK_TYPES.LOCALHOST:
-        return t('localhost');
-      default:
-        return provider.nickname || t('unknownNetwork');
-    }
-  }
-
```
Comment (digiwand) [inline ui/components/app/dropdowns/network-dropdown.js:250]: nice cleanup find

File: ui/components/app/signature-request-siwe/signature-request-siwe.js [hunk 3]
```diff
@@ -30,6 +44,10 @@ export default function SignatureRequestSIWE({
 }) {
   const allAccounts = useSelector(accountsWithSendEtherInfoSelector);
   const subjectMetadata = useSelector(getSubjectMetadata);
+  const nativeCurrency = useSelector(getNativeCurrency);
+  const currentCurrency = useSelector(getCurrentCurrency);
+  const currentChainId = useSelector(getCurrentChainId);
+  const provider = useSelector(getProvider);
```
Comment (digiwand) [inline ui/components/app/signature-request-siwe/signature-request-siwe.js:50]: [out-of-scope] interesting...  wonder how much of this logic could be moved into `NetworkAccountBalanceHeader` itself

File: ui/components/app/signature-request-siwe/signature-request-siwe.js [hunk 4]
```diff
@@ -84,8 +102,46 @@ export default function SignatureRequestSIWE({
...
+      <NetworkAccountBalanceHeader
+        networkName={currentNetwork}
+        accountName={fromAccount.name}
+        accountBalance={balanceInBaseAsset}
+        tokenName={
+          conversionRate ? currentCurrency?.toUpperCase() : nativeCurrency
+        }
+        accountAddress={fromAccount.address}
+        chainId={currentChainId}
+      />
```
Comment (digiwand) [inline ui/components/app/signature-request-siwe/signature-request-siwe.js:144]: what do you think about moving the logic into the Header component from './signature-request-siwe-header' to maintain separations of concerns?

**Pull request closed without merging**